internal/datastore/crdb/readwrite.go

```diff
@@ -104,6 +113,50 @@ var (
 	queryDeleteCounter = psql.Delete(tableRelationshipCounter)
 )
 
+func (rwt *crdbReadWriteTXN) insertQuery() sq.InsertBuilder {
+	return psql.Insert(rwt.tupleTableName)
+}
+
+func (rwt *crdbReadWriteTXN) queryDeleteTuples() sq.DeleteBuilder {
+	return psql.Delete(rwt.tupleTableName)
+}
+
+func (rwt *crdbReadWriteTXN) queryWriteTuple() sq.InsertBuilder {
+	if rwt.withIntegrity {
```

Comment:
sorry for the late note - can we remove these ifs from every call? i.e. pick a different implementation of `crdbReadWriteTXN` on initialization instead?

Reply:
That would be a lot of copy-paste of code and reducing reuse a lot